src/MICore/Debugger.cs

```diff
@@ -578,20 +578,23 @@ public Task CmdBreak(BreakRequest request)
             return CmdBreakInternal();
         }
 
-        internal bool IsLocalGdb()
+        protected bool IsLocalLaunchUsingServer()
         {
-            return (this.MICommandFactory.Mode == MIMode.Gdb &&
-               this._launchOptions is LocalLaunchOptions &&
-               String.IsNullOrEmpty(((LocalLaunchOptions)this._launchOptions).MIDebuggerServerAddress));
+            return (_launchOptions is LocalLaunchOptions &&
```

Comment:
nit: C# now has a better way of writing this code. Since we are touching it, we might as well use it:

```
            return (_launchOptions is LocalLaunchOptions localLaunchOptions &&
                (!String.IsNullOrWhiteSpace(localLaunchOptions.MIDebuggerServerAddress) ||
                 !String.IsNullOrWhiteSpace(localLaunchOptions.DebugServer));
```

Reply:
Ohh nice!! Did not know that. Done, will wait for more comments before another round of testing.

Reply:
Done, that is such a nice feature, bit obscure syntax -- declaring a variable in the middle of an expression -- but the scope of that var was reasonable. Thanks for the education.